src/diffusers/schedulers/scheduling_ddim.py

```diff
@@ -296,9 +296,15 @@ def step(
                 )
 
             if variance_noise is None:
-                variance_noise = torch.randn(model_output.shape, dtype=model_output.dtype, generator=generator).to(
-                    device
-                )
+                if variance_noise is None:
+                    if device.type == "mps":
```

Comment:
```suggestion
                    if str(device) == "mps":
```

The device can be a string ("cpu" above)